app/frontend/src/features/communities/leaveReference/LeaveReferencePage.test.tsx

```diff
@@ -0,0 +1,187 @@
+import { render, screen, within } from "@testing-library/react";
+import {
+  INVALID_REFERENCE_TYPE,
+  REFERENCE_TYPE_NOT_AVAILABLE,
+} from "features/communities/constants";
+import { Route } from "react-router-dom";
+import { leaveReferenceBaseRoute } from "routes";
+import { service } from "service";
+import { getHookWrapperWithClient } from "test/hookWrapper";
+import { getAvailableReferences, getUser } from "test/serviceMockDefaults";
+import { MockedService } from "test/utils";
+
+import LeaveReferencePage from "./LeaveReferencePage";
+
+const getAvailableReferencesMock = service.references
+  .getAvailableReferences as MockedService<
+  typeof service.references.getAvailableReferences
+>;
+const getUserMock = service.user.getUser as MockedService<
+  typeof service.user.getUser
+>;
+
+function renderLeaveFriendReferencePage(referenceType: string, userId: string) {
+  const { wrapper } = getHookWrapperWithClient({
+    initialRouterEntries: [
+      `${leaveReferenceBaseRoute}/${referenceType}/${userId}`,
+    ],
+  });
+
+  render(
+    <Route path={`${leaveReferenceBaseRoute}/:referenceType/:userId`}>
+      <LeaveReferencePage />
+    </Route>,
+    { wrapper }
+  );
+}
+
+function renderLeaveRequestReferencePage(
+  referenceType: string,
+  userId: string,
+  hostRequest: number
+) {
+  const { wrapper } = getHookWrapperWithClient({
+    initialRouterEntries: [
+      `${leaveReferenceBaseRoute}/${referenceType}/${userId}/${hostRequest}`,
+    ],
+  });
+
+  render(
+    <Route
+      path={`${leaveReferenceBaseRoute}/:referenceType/:userId/:hostRequest`}
+    >
+      <LeaveReferencePage />
+    </Route>,
+    { wrapper }
+  );
+}
+
+describe("LeaveReferencePage", () => {
+  beforeEach(() => {
+    getUserMock.mockImplementation(getUser);
+    getAvailableReferencesMock.mockImplementation(getAvailableReferences);
+  });
+
+  describe("When the reference type is invalid", () => {
+    beforeEach(() => {
+      renderLeaveFriendReferencePage("hello", "1");
+    });
+
+    it("Returns an error", async () => {
+      const errorAlert = await screen.findByRole("alert");
+      expect(
+        within(errorAlert).getByText(INVALID_REFERENCE_TYPE)
+      ).toBeVisible();
+    });
+
+    it("does not show the form", () => {
+      expect(
+        screen.queryByRole("heading", {
+          name: "You met with Friendly Cow",
+        })
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  describe("When the reference type is friend", () => {
+    describe("and the users are friends", () => {
+      beforeEach(() => {
+        renderLeaveFriendReferencePage("friend", "5");
+      });
+
+      it("verifies that the review type is available", async () => {
+        expect(getAvailableReferencesMock).toHaveBeenCalledTimes(1);
+        expect(getAvailableReferencesMock).toHaveReturned();
```

Comment:
```suggestion
        expect(getAvailableReferencesMock).toHaveBeenCalledWith({ userId: 5 });
```
I think this is a more valuable assertion (that the function is called with the right input, then we can assume the function does the right thing if we call with the right input) as opposed to asserting the function has returned, as if not, the test would have crashed with an error/failed anyway